### ISSUE TITLE
Character escapes in jsonpath::json_location::to_string

### ISSUE THREAD

**Issue by gerdy-d** [danielaparker/jsoncons#410]
**Describe the bug**

It seems that `json_location::to_string` (jsonpath) only escapes single quotes (and no other characters) within a  `json_location_node_kind::name`.

**Enumerate the steps to reproduce the bug**

Please see the example below.

**Include a small, self-contained example if possible**

Consider the following example program (#includes omitted):

```
    const json j = json::parse(LR"({"\\":0})");
    const std::wstring p = LR"($['\\'])";
    jsonpath::json_query(j,
                         p,
                         [](const json::string_view_type& normalized_path, const json&) {
                             std::wcout << normalized_path;
                         },
                         jsonpath::result_options::path);
```

This prints:
`$['']`

I'm not entirely sure, but I would expect the program to print
`$['\\']`

Using
`jsoncons::jsonpath::escape_string(node->name().c_str(), node->name().length(), buffer);`
at [lines 331-342](https://github.com/danielaparker/jsoncons/blob/master/include/jsoncons_ext/jsonpath/json_location.hpp#L331) seems to result in the output `$['\\']`, but I'm uncertain if this modification in the code would be correct.

Thanks in advance for looking into this.

**What compiler, architecture, and operating system?**

- Compiler: Visual C++ 2019
- Architecture (e.g. x86, x64) x86
- Operating system: Windows 10

**What jsoncons library version?**

 - [X] Latest release 0.169.0
 - [ ] Other release ______
 - [ ] master

**Comment by danielaparker** [comment]
Regarding your example, I'm pretty sure we don't support that combination of narrow characters and wide characters!

But yes, you're right,

**Comment by danielaparker** [comment]
Fix is on master. This code

```
std::string json_string = R"({"\\":0})";
std::cout << "json string: " << json_string << "\n\n";

const json j = json::parse(json_string);

const std::string p = R"($['\\'])";
jsonpath::json_query(j,
    p,
    [j](const json::string_view_type& normalized_path, const json&) {
        auto value = jsonpath::json_query(j, normalized_path);
        std::cout << "normalized_path: " << normalized_path << ", value: " << value << "\n";
    },
    jsonpath::result_options::path);
```

produces

```
json string: {"\\":0}

normalized_path: $['\\'], value: [0]
```

**Comment by gerdy-d** [comment]
Sorry for the confusion regarding narrow and wide characters, I forgot to paste this line in my example:
`using json = jsoncons::wojson;`

Thanks for the fix!

**Comment by danielaparker** [comment]
Fix is in 0.170.0
